chore: add breaking changes to end of PR template (#28941)
aws/aws-cdk: .github/PULL_REQUEST_TEMPLATE.md

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

<!--Does this PR includes breaking changes? If it does, list them here in the following format (notice how multiple breaking changes should be formatted):

  ```

>   BREAKING CHANGE: Description of what broke and how to achieve this behavior now

Reviewer: ```suggestion
  BREAKING CHANGE: Description of the breaking change, the previous behaviour before and the new behaviour. 
```
